c/misra/test/rules/RULE-21-13/test.c

```diff
@@ -0,0 +1,26 @@
+#include <ctype.h>
+#include <stdio.h>
+
```

Comment:
```c
int c = getchar(); // only returns -1 or a unsigned char
isdigit(c); // COMPLIANT[FALSE_POSITIVE]
```

Then this could be supported using a [`SimpleRangeAnalysisExpr`](https://github.com/github/codeql/blob/7dc5e7e0fe4ac147af4bcdc02830a47a673f517c/cpp/ql/lib/experimental/semmle/code/cpp/models/interfaces/SimpleRangeAnalysisExpr.qll#L16) to define the values provided by `getchar`, something like:

```codeql
class GetCharSimpleRangeAnalysisExpr extends SimpleRangeAnalysisExpr, FunctionCall {
  GetCharSimpleRangeAnalysisExpr() {
    this.getTarget().hasName("getchar")
  }

  override float getLowerBounds() { result = -1 }

  override float getUpperBounds() { result = 255 }
}
```
Added to https://github.com/github/codeql-coding-standards/blob/main/cpp/common/src/codingstandards/cpp/SimpleRangeAnalysisCustomizations.qll.

Reply:
Add a test case for this as well:
```
int y = 0;
if (x == 1) {
  y = EOF;
} else {
  y = 'c';
}
isdigit(y);
```

Reply:
`isdigit(EOF);`

Reply:
Added both. In the case of modelling getchar, also added friends of `getchar` that also returns values within 1) unsigned char and 2) EOF.